node/src/accountant/payable_dao.rs

```diff
@@ -582,22 +682,72 @@ mod tests {
             .unwrap();
         let subject = PayableDaoReal::new(conn);
 
-        let _ = subject.mark_pending_payable_rowid(&wallet, rowid);
+        let _ = subject.mark_pending_payables_rowids(&[(&wallet, rowid)]);
     }
 
     #[test]
-    fn mark_pending_payment_handles_general_sql_error() {
+    #[should_panic(
+        expected = "Marking pending payable rowid for wallets 0x000000000000000000000000000000686f6f6761, \
+         0x000000000000000000000000000000626f6f6761, 0x00000000000000000000626f6f6761686f6f6761 affected 1 rows but expected 3. \
+         Accounts for wallets (0x000000000000000000000000000000626f6f6761, 0x00000000000000000000626f6f6761686f6f6761) had contained \
+          rowids when we tried to update them with new ones. All such columns should be emptied beforehand at the confirmation of the \
+           earlier transactions. A malformed, repeated payment is suspected"
+    )]
+    fn mark_pending_payables_rowids_refuses_to_overwrite_existing_marked_rowids() {
         let home_dir = ensure_node_home_directory_exists(
             "payable_dao",
-            "mark_pending_payment_handles_general_sql_error",
+            "mark_pending_payables_rowids_refuses_to_overwrite_existing_marked_rowids",
+        );
+        let wallet_1 = make_wallet("hooga");
+        let rowid_1 = 550;
+        let wallet_2 = make_wallet("booga");
+        let rowid_2 = 555;
+        let wallet_3 = make_wallet("boogahooga");
+        let rowid_3 = 558;
+        let conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        insert_record_fn(&*conn, &wallet_1.to_string(), 12345, 1_000_000_000, Some(1));
+        conn.prepare("update payable set pending_payable_rowid = null where wallet_address = ?")
```

Comment:
Is this necessary? Can you call the function above differently and achieve the same result?